server/raft.go

```diff
@@ -3383,7 +3383,7 @@ func (n *raft) processAppendEntry(ae *appendEntry, sub *subscription) {
 
 			if n.commit > 0 && ae.pindex <= n.commit {
 				// Check if only our terms do not match here.
-				if ae.pindex == n.pindex {
+				if ae.pindex == n.pindex || ae.pindex == n.commit {
```

Comment:
I'm not following the logic here. This case may just reset the `pterm` incorrectly?